apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -278,41 +280,67 @@ export const submitChatContents = createAsyncThunk(
       currentSessionId
     );
 
-    thunkAPI.dispatch(setChatSessionId(chatApiResponse.session_id));
-    thunkAPI.dispatch(
-      updateChatMessageSession({
-        id: newMessage.id,
-        sessionId: chatApiResponse.session_id,
-      })
-    );
-
-    if (chatApiResponse?.status === 'profanity') {
-      // Logging to allow visibility into flagged content.
-      console.log(chatApiResponse);
-
-      return thunkAPI.dispatch(
-        updateUserChatMessageStatus({
+    // Regardless of response type,
+    // assign last user message to session.
+    if (chatApiResponse.session_id) {
+      thunkAPI.dispatch(setChatSessionId(chatApiResponse.session_id));
+      thunkAPI.dispatch(
+        updateChatMessageSession({
           id: newMessage.id,
-          status: Status.PROFANITY_VIOLATION,
+          sessionId: chatApiResponse.session_id,
         })
       );
     }
 
+    // success state: received response from model ("assistant")
     if (chatApiResponse?.role === Role.ASSISTANT) {
       const assistantChatMessage: ChatCompletionMessage = {
         id: getNewMessageId(),
         role: Role.ASSISTANT,
         status: Status.OK,
         chatMessageText: chatApiResponse.content,
-        // The accuracy of this timestamp is debatable since it's not when our backend
-        // issued the message, but it's good enough for user testing.
         timestamp: getCurrentTimestamp(),
         sessionId: chatApiResponse.session_id,
       };
-      return thunkAPI.dispatch(addChatMessage(assistantChatMessage));
-    }
+      thunkAPI.dispatch(addChatMessage(assistantChatMessage));
```

Comment:
This I think is out of scope for this PR, but wanted to get any thoughts other folks have...I chatted with Sanchit a little bit about this, but I'm getting anxious about the degree to which logging currently relies on keeping the state of chat messages in sync between what's in Redux and what's being stored in our logging model in Rails.

For example, we rely on this `assistantChatMessage` being added to the chat history in a way that matches the chat message we've added to the session in Rails here:

https://github.com/code-dot-org/code-dot-org/blob/1f8d78272f3897668876df0bc5a309a7abde61b9/dashboard/app/controllers/aichat_controller.rb#L51-L55

I don't have this fully thought out, I'm wondering if we could pass back and forth the entire list of messages instead of doing this appending on both sides (and relying on that appending to be the same).

Reply:
What exactly is the concern? It sounds like the worry is that we have a list of messages on the client in redux, and we're trying to propagate that same list of messages to the server, _and_ that the list could potentially get out of sync. So the speculation is maybe to send the entire list of chat messages. Am I following it?

If so, maybe just sending a unique incrementing ID could do it. Maybe we namespace it with some unique value?

So when we start up, we assign that we're dealing with chat `X` (or whatever) and then we send along an incrementing id `13` for this message - so we tack into the packet `messageID: 'X,13'` (or however it's formatted. Maybe they're fixed width).

Then the server keeps track of if we've see the previous message. If so, cool - we're gtg. If not, then it fails back to the client with a message of the _last_ ID it's successfully received. The client then transmits the next ID up through the present. If the server receives a duplicate message ID, it just discards and ignores it (maybe the other request shows up late).

It's a little more work to implement, but it keeps us still sending only the updated messages, and re-transmitting only the missed ones if anything's dropped.

Or maybe I'm misunderstanding the issue. 😄 

Regardless, I do agree with the sentiment of it being out of scope for this PR, so we should prob bump to a slack thread to discuss further.

Reply:
Thanks for the thoughts y'all -- just set a reminder to start a Slack thread on this on Monday :)